python/orca/src/bigdl/orca/learn/pytorch/experimential/mmcv/mmcv_ray_runner.py

```diff
@@ -14,19 +14,89 @@
 # limitations under the License.
 #
 
+import io
+import os
+import tempfile
 import time
 import mmcv
 import numpy as np
 import torch
 import warnings
+from collections import OrderedDict
 from mmcv.runner import EpochBasedRunner
 from mmcv.runner.utils import get_host_info
 from mmcv.parallel.distributed import MMDistributedDataParallel
+from mmcv.parallel.utils import is_module_wrapper
+from mmcv import BaseStorageBackend, FileClient
+from mmcv.runner.checkpoint import CheckpointLoader
 from bigdl.orca.learn.pytorch.utils import get_batchsize
 from bigdl.dllib.utils.log4Error import invalidInputError
 from bigdl.orca.learn.pytorch.experimential.core.base_ray_runner import BaseRayRunner
 
-from typing import (Any, Dict, List, Optional, Tuple, Callable, overload)
+from typing import (Union, Any, Dict, List, Optional, Tuple, Callable, overload)
+
+
+class HDFSBackend(BaseStorageBackend):
+    """
+    HDFS storage backend for saving ckpt
+
+    This backend will be used when runner contains a CheckpointHook and
+    CheckpointHook's out_dir starts with hdfs://
+    """
+
+    def get(self, filepath: str) -> bytes:
+        pass
+
+    def get_text(self, filepath: str, encoding: str = 'utf-8') -> str:
+        pass
+
+    def put(self, obj: bytes, filepath: str) -> None:
+        filename = os.path.basename(filepath)
+        temp_dir = tempfile.mkdtemp()
+        temp_path = os.path.join(temp_dir, filename)
+        with open(temp_path, "wb") as f:
+            f.write(obj)
+
+        from bigdl.orca.data.file import exists, makedirs, put_local_file_to_remote
+        work_dir = os.path.dirname(filepath)
+        if not exists(work_dir):
+            makedirs(work_dir)
+        put_local_file_to_remote(temp_path, filepath)
+
+    def put_text(self, obj: str, filepath: str, encoding: str = 'utf-8') -> None:
+        pass
```

Comment:
involved by `mmcv.dump`, maybe you don't need to fix this in this pr, but set a reminder here.